File: C/1021 - Notas e Moedas.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

int main() {
    double grossTotal, sum1, sum2;
    int nota, restoCem, restoCinquenta, restoVinte, restoDez, restoCinco, sum3, restoReal, restoCinquentaCentavo, restoVinteCincoCentavo, restoDezCentavo;
    int cem, cinquenta, vinte, dez, cinco, dois;
    int umReal, cinquentaCentavos, vinteCincoCentavos, dezCentavos, cincoCentavos, umCentavo;

    scanf ("%lf", &grossTotal);

    nota = grossTotal;
    cem = nota / 100;
    restoCem = nota % 100;
    cinquenta = restoCem / 50;
    restoCinquenta = restoCem % 50;
    vinte = restoCinquenta / 20;
    restoVinte = restoCinquenta % 20;
    dez = restoVinte / 10;
    restoDez = restoVinte % 10;
    cinco = restoDez / 5;
    restoCinco = restoDez % 5;
    dois = restoCinco / 2;
    umReal = restoCinco % 2;

    sum2 = grossTotal * 100;
    sum3 = (int) sum2;
    restoReal = sum3 % 100;
    cinquentaCentavos = restoReal / 50;
    restoCinquentaCentavo = restoReal % 50;
    vinteCincoCentavos = restoCinquentaCentavo / 25;
    restoVinteCincoCentavo = restoCinquentaCentavo % 25;
    dezCentavos = restoVinteCincoCentavo / 10;
    restoDezCentavo = restoVinteCincoCentavo % 10;
    cincoCentavos = restoDezCentavo / 5;
    umCentavo = restoDezCentavo % 5;

    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n",cem);
    printf("%d nota(s) de R$ 50.00\n",cinquenta);
    printf("%d nota(s) de R$ 20.00\n",vinte);
    printf("%d nota(s) de R$ 10.00\n",dez);
    printf("%d nota(s) de R$ 5.00\n",cinco);
    printf("%d nota(s) de R$ 2.00\n",dois);
    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n",umReal);
    printf("%d moeda(s) de R$ 0.50\n",cinquentaCentavos);
    printf("%d moeda(s) de R$ 0.25\n",vinteCincoCentavos);
    printf("%d moeda(s) de R$ 0.10\n",dezCentavos);
    printf("%d moeda(s) de R$ 0.05\n",cincoCentavos);
    printf("%d moeda(s) de R$ 0.01\n",umCentavo);

    return 0;
}

File: PYTHON/1021 - Notas e Moedas.py
n = float(input())

n100 = n // 100
n = n - n100*100

n50 = n // 50
n = n - n50*50

n20 = n // 20
n = n - n20*20

n10 = n // 10
n = n - n10*10

n5 = n // 5
n = n - n5*5

n2 = n // 2
n = n - n2*2

n1 = n // 1
n = n - n1*1
n1=float('%.2f'% n1)
n=float('%.2f'% n)

m50 = n // 0.5
n = n - m50*0.5
m50=float('%.2f'% m50)
n=float('%.2f'% n)

m25 = n // 0.25
n = n - m25*0.25
m25=float('%.2f'% m25)
n=float('%.2f'% n)

m10 = n // 0.10
n = n - m10*0.10
m10=float('%.2f'% m10)
n=float('%.2f'% n)

m5 = n // 0.05
n = n - m5*0.05
m5=float('%.2f'% m5)
n=float('%.2f'% n)

m1 = n * 100
m1=float('%.2f'% m1)
n=float('%.2f'% n)

print('NOTAS:')
print(f'{n100:.0f} nota(s) de R$ 100.00')
print(f'{n50:.0f} nota(s) de R$ 50.00')
print(f'{n20:.0f} nota(s) de R$ 20.00')
print(f'{n10:.0f} nota(s) de R$ 10.00')
print(f'{n5:.0f} nota(s) de R$ 5.00')
print(f'{n2:.0f} nota(s) de R$ 2.00')
print(f'MOEDAS:')
print(f'{n1:.0f} moeda(s) de R$ 1.00')
print(f'{m50:.0f} moeda(s) de R$ 0.50')
print(f'{m25:.0f} moeda(s) de R$ 0.25')
print(f'{m10:.0f} moeda(s) de R$ 0.10')
print(f'{m5:.0f} moeda(s) de R$ 0.05')
print(f'{m1:.0f} moeda(s) de R$ 0.01')